src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php

```diff
@@ -251,6 +251,17 @@ public function getRedirectParameters($resource = null)
 
         $parameters = isset($redirect['parameters']) ? $redirect['parameters'] : [];
 
+        $vars = $this->getVars();
+        $accessor = PropertyAccess::createPropertyAccessor();
+
+        if ($accessor->isReadable($vars, '[redirect][parameters]')) {
+            $extraParameters = $accessor->getValue($vars, '[redirect][parameters]');
+
+            if (is_array($extraParameters)) {
+                $parameters = array_merge($parameters, $extraParameters);
+            }
+        }
```

Comment:
To extend the readability and keep functions relatively small, what would you say for extracting this to the private method?